### DISCUSSION TITLE
How to convert a table row into a list

### DISCUSSION THREAD

**Discussion by raffomania** [nushell/nushell#4148] [Q&A] [answered]
I'm trying to do something like this:
```
echo "12apples 25bananas | parse "{apples}apples {bananas}bananas" | <conversion of row into list> | each { $it | into int }
```
So I can do arithmetics on the amount of apples and bananas. Which is the missing command?

Another way to put the question: How do I run a command for every column of a row?

**Comment by jbertovic** [comment, marked as answer]
When you parse the string you end up getting two columns of strings.  One way would be to convert the columns to integer.  I always use the command describe to understand what I'm working with.

This will show you that they are parsed as strings
`echo "12apples 25bananas" | parse "{apples}apples {bananas}bananas" | describe`

Then you can convert the columns into integer.  You could run describe on this to see the outcome
`echo "12apples 25bananas" | parse "{apples}apples {bananas}bananas" | into int apples bananas`

Then to run on each row you use the command `each`. `$it` indicates item/iterator meaning row here.
`echo "12apples 25bananas" | parse "{apples}apples {bananas}bananas" | into int apples bananas | each { echo $"you have (($it.apples) + ($it.bananas)) fruit"}`

**Reply by raffomania** [reply]
Thanks, passing multiple column names to `into int` was what I was missing!

**Reply by Xevion** [reply]
~~(3 years later, I know)
This answer shows up now as the top answer, and while your answer was correct in the spirit of the question (classic XY problem)... It unfortunately is completely unhelpful for the name of the question. It doesn't help me, since I'm not converting integers.~~

Offered my own answer [here](https://github.com/nushell/nushell/discussions/4148#discussioncomment-11077670).

**Comment by Xevion** [comment]
It may seem odd to reply to this 3 years after it was answered, but since it's a top Google result and the current answer is not directly pertinent to the question's title, I'm providing my own variant.

The most apt choice would be the `values` command. See here:

```nu
$ ["test -d /mnt/D/DCIM", "test -r /mnt/D/DCIM"] | split column ' '
╭───┬─────────┬─────────┬─────────────╮
│ # │ column1 │ column2 │   column3   │
├───┼─────────┼─────────┼─────────────┤
│ 0 │ test    │ -d      │ /mnt/D/DCIM │
│ 1 │ test    │ -r      │ /mnt/D/DCIM │
╰───┴─────────┴─────────┴─────────────╯

$ ["test -d /mnt/D/DCIM", "test -r /mnt/D/DCIM"] | split column ' ' | each {values}
╭───┬─────────────────────╮
│ 0 │ ╭───┬─────────────╮ │
│   │ │ 0 │ test        │ │
│   │ │ 1 │ -d          │ │
│   │ │ 2 │ /mnt/D/DCIM │ │
│   │ ╰───┴─────────────╯ │
│ 1 │ ╭───┬─────────────╮ │
│   │ │ 0 │ test        │ │
│   │ │ 1 │ -r          │ │
│   │ │ 2 │ /mnt/D/DCIM │ │
│   │ ╰───┴─────────────╯ │
╰───┴─────────────────────╯
```

Still new to this language, so if there's a technically superior method of achieving this result that is better - I am unaware.

**Reply by weirdan** [reply]
`values` is the correct answer to convert a record (all table rows are records) into a list. Depending on whether only the first row is needed (as in the original question) or all rows should be converted (resulting in list of lists) it would be either `... | first | values | ...` or `... | each { values } | ... ` (`$in` is optional).

**Reply by Xevion** [reply]
> or `... | each { values } | ... ` (`$in` is optional).

Awesome, I didn't know I could do that. I've updated my answer to reflect this terse syntax.
